commit 5188eeeaa673e45a22e9433bb596fef1a16e56d6
Author: San Mehat <san@google.com>
Date:   Wed Jul 29 20:21:28 2009 -0700

    serial_core: Add wake_peer uart operation
    
    Add wake_peer which is called before starting UART TX. The idea here
    is to provide a mechanism where we can wakeup our peer before sending
    data.
    
    Change-Id: I42e0779b635f64ca99184b45d5b028de80197491
    Signed-off-by: San Mehat <san@google.com>

diff --git a/drivers/tty/serial/serial_core.c b/drivers/tty/serial/serial_core.c
index 9c4c05b..7d9fbb8 100644
--- a/drivers/tty/serial/serial_core.c
+++ b/drivers/tty/serial/serial_core.c
@@ -94,6 +94,9 @@ static void __uart_start(struct tty_struct *tty)
 	struct uart_state *state = tty->driver_data;
 	struct uart_port *port = state->uart_port;
 
+	if (port->ops->wake_peer)
+		port->ops->wake_peer(port);
+
 	if (!uart_circ_empty(&state->xmit) && state->xmit.buf &&
 	    !tty->stopped && !tty->hw_stopped)
 		port->ops->start_tx(port);
diff --git a/include/linux/serial_core.h b/include/linux/serial_core.h
index f51bf2e..046d895 100644
--- a/include/linux/serial_core.h
+++ b/include/linux/serial_core.h
@@ -252,6 +252,7 @@ struct uart_ops {
 	void		(*pm)(struct uart_port *, unsigned int state,
 			      unsigned int oldstate);
 	int		(*set_wake)(struct uart_port *, unsigned int state);
+	void		(*wake_peer)(struct uart_port *);
 
 	/*
 	 * Return a string describing the type of the port
